### ISSUE TITLE
XYCharts with incorrect type results in infinite loop 

### ISSUE THREAD

**Issue by Gervwyk** [amcharts/amcharts4#1320]
When changing a series type value to something incorrect, using json config, the chart results in infinite loop. 
Check out the sandbox example:
https://codesandbox.io/s/amcharts4-type-inf-loop-error-j9huk
Change the chartConfig.js to:

```
const chartConfig = {
  fontSize: "10px",
  xAxes: [
    {
      type: "CategoryAxis",
      title: {
        text: "Countries"
      },
      dataFields: {
        category: "country",
        title: {
          text: "Countries"
        }
      }
    }
  ],
  yAxes: [
    {
      type: "ValueAxis",
      title: {
        text: "Litres sold (M)"
      }
    }
  ],
  series: [
    {
      // type: "ColumnSeries", // change this line to Series and it will cause infinate loop
      type: "Series",
      dataFields: {
        valueY: "litres",
        categoryX: "country"
      },
      name: "Sales",
      columns: {
        tooltipText: "Series: {name}\nCategory: {categoryX}\nValue: {valueY}",
        stroke: "#6871dc",
        fill: "#67b6dc"
      }
    },
    {
      type: "LineSeries",
      stroke: "#db6788",
      strokeWidth: 3,
      dataFields: {
        valueY: "units",
        categoryX: "country"
      },
      name: "Units"
    }
  ]
};

export default chartConfig;
```

**Comment by martynasma** [comment]
Why would you do what you know is definitely wrong? `Series` is a base class, not meant for use.

**Comment by Gervwyk** [comment]
A user can define these chart configs in my use case. So its just a bad experience if they make a config mistake and the app stops responding, which will happen in this case.
I tried catching the error in the react component, which should solve the problem, but was unsuccessful. 
Any idea how to stop this from happening?

**Comment by martynasma** [comment]
Not sure :)

I'm going to take a look whether we can do something on our end, not let it fail silently in obviously faulty configurations like this.

**Comment by Gervwyk** [comment]
Great, thanks! 
It would also be great if there was a way to 'sanitize' the json config, or just a single schema description for the json config which include all allowed values.

**Comment by martynasma** [comment]
Next version will error out if you try to use `Series`.

However, it's only for specific case. There might be other cases where use of incorrect classes might end up in dead loop.

BTW, it's not JSON-specific error. If you would instantiate `Series` and use it in object-based setup, you'd end up with the same issue.

**Comment by Gervwyk** [comment]
Excellent! Thanks. Btw, I found `Column` and `Line` also does the same. Not sure what others there might be or which other field than `type` would cause the same issue.

**Comment by martynasma** [comment]
Sorry I jumped the gun. Looks like the change has effect on other things so we have reverted the change for now.

Will keep this open for now.

**Comment by martynasma** [comment]
Fixed in [4.4.9](https://github.com/amcharts/amcharts4/releases/tag/4.4.9).

## [4.4.9] - 2019-05-28

### Added
- `startIndex` property added to `ColorSet`. This allows to specify from which color charts like `PieChart` should start when picking colors for slices.
- `excludeFromTotal` property added to `XYSeries` (default: `false`). Allows excluding values of particular series when caluclating totals.
- `includeRangesInMinMax` property added to `ValueAxis`.

### Changed
- The chart will now error out if trying to instantiate generic classes like  (`Chart`, `Axis`, `Series`).
- Legend in `TreeMap` was completely revamped. Previously it used series as data, but as series were not always created, it failed to render items in some cases. Now it uses `TreeMapDataItem` of `TreeMap`.
- `TreeMap` legend will now show items of first level which has more than one data item.
- `dataContext` of `ForceDirectedNodeDataItem` will now have the same object as target node's.

### Fixed
- `responsive.useDefault = false` was being ignored.
- `DateAxis` with hourly-based data could render some of the labels using date formatter of a changed period even if the period didn't actually change.
- In some cases `MapChart` could interfere with event objects of other charts on the same page, even when not directly interacting with it.
- If `baseInterval` of `DateAxis` didn't have `count` set, the chart failed to render.
- Setting data on hidden Series resulted tooltips on invisible bullets to be shown.
- `WordCloudSeries` with single label or several labels with the same values was displaying labels using `minFontSize`.
- `axis.createSeriesRange()` was not working with `RadarColumnSeries`.
- When data was added to a chart, series could flicker at incorrect position before rendering properly. Also the same could happen when zooming chart (it was only visible if animated theme was not in use).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
